activerecord/lib/active_record/core.rb

```diff
@@ -413,7 +413,7 @@ def encode_with(coder)
     # models are still comparable.
     def ==(comparison_object)
       super ||
-        comparison_object.instance_of?(self.class) &&
+        comparison_object.is_a?(self.class.base_class) &&
```

Comment:
Do we want to support `STISuperClass.first == STISubClass.first`? You would need to do something like `comparison_object.class.base_class == self.class.base_class` to match equality in both combinations.

Reply:
I don't think we need to move to `comparison_object.class.base_class == self.class.base_class` in order to work for both comparisons.  The comparison object in both cases is going to be a subclass of the top level STI base class.  So the `is_a?` operation should always work.  In your example `self.class.base_class` will be `STISuperClass` and `STISuperClass.first.is_a? STISuperClass` and `STISubClass.first.is_a? STISuperClass` will both be true.

Reply:
Ah! You are correct. I thought `#is_a?` and `#instance_of?` were synonymous, but they are not. 👍 In that case please rebase and squash your commits for this to be mergeable. Also I think this is CHANGELOG worthy.